plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,340 @@
+	formValues := map[string]string{
+		"gwos-app-name":  "telegraf",
+		"gwos-api-token": g.authToken,
+	}
+
+	headers := map[string]string{
+		"Content-Type": "application/x-www-form-urlencoded",
+	}
+
+	_, _, err := clients.SendRequest(http.MethodPost, g.Server+logoutURL, headers, formValues, nil)
+
+	return err
+}
+
+func (g *Groundwork) Write(metrics []telegraf.Metric) error {
+	resourceToServicesMap := make(map[string][]transit.DynamicMonitoredService)
+	for _, metric := range metrics {
+		resource, service := parseMetric(g.DefaultHost, g.DefaultServiceState, g.ResourceTag, metric)
+		resourceToServicesMap[resource] = append(resourceToServicesMap[resource], service)
+	}
+
+	var resources []transit.DynamicMonitoredResource
+	for resourceName, services := range resourceToServicesMap {
+		resources = append(resources, transit.DynamicMonitoredResource{
+			BaseResource: transit.BaseResource{
+				BaseTransitData: transit.BaseTransitData{
+					Name: resourceName,
+					Type: transit.Host,
+				},
+			},
+			Status:        transit.HostUp,
+			LastCheckTime: milliseconds.MillisecondTimestamp{Time: time.Now()},
+			NextCheckTime: milliseconds.MillisecondTimestamp{Time: time.Now()},
+			Services:      services,
+		})
+	}
+
+	traceToken, _ := uuid.GenerateUUID()
+	requestJSON, err := json.Marshal(transit.DynamicResourcesWithServicesRequest{
+		Context: &transit.TracerContext{
+			AppType:    "TELEGRAF",
+			AgentID:    g.AgentID,
+			TraceToken: traceToken,
+			TimeStamp:  milliseconds.MillisecondTimestamp{Time: time.Now()},
+			Version:    transit.ModelVersion,
+		},
+		Resources: resources,
+		Groups:    nil,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	headers := map[string]string{
+		"GWOS-APP-NAME":  "telegraf",
+		"GWOS-API-TOKEN": g.authToken,
+		"Content-Type":   "application/json",
+		"Accept":         "application/json",
+	}
+
+	statusCode, body, err := clients.SendRequest(http.MethodPost, g.Server+defaultMonitoringRoute, headers, nil, requestJSON)
+	if err != nil {
+		return err
+	}
+
+	/* Re-login mechanism */
+	if statusCode == 401 {
+		if err = g.Connect(); err != nil {
+			return err
+		}
+		headers["GWOS-API-TOKEN"] = g.authToken
+		statusCode, body, err = clients.SendRequest(http.MethodPost, g.Server+defaultMonitoringRoute, headers, nil, requestJSON)
+		if err != nil {
+			return err
+		}
+	}
+
+	if statusCode != 200 {
+		return fmt.Errorf("something went wrong during processing an http request[http_status = %d, body = %s]", statusCode, string(body))
+	}
+
+	return nil
+}
+
+func (g *Groundwork) Description() string {
+	return "Send telegraf metrics to GroundWork Monitor"
+}
+
+func init() {
+	outputs.Add("groundwork", func() telegraf.Output {
+		return &Groundwork{}
+	})
+}
+
+func parseMetric(defaultHostname, defaultServiceState, resourceTag string, metric telegraf.Metric) (string, transit.DynamicMonitoredService) {
+	resource := defaultHostname
+
+	if value, present := metric.GetTag(resourceTag); present {
```

Comment:
```suggestion

	resource := defaultHostname
	if value, present := metric.GetTag(resourceTag); present {
```
Let's keep them together..

Reply:
Done